modules/cart/commerce_cart.module

```diff
@@ -122,6 +122,44 @@ function commerce_cart_form_commerce_order_type_form_alter(array &$form, FormSta
 }
 
 /**
+ * Implements hook_form_FORM_ID_alter() for 'field_ui_field_edit_form'.
+ */
+function commerce_order_form_field_config_edit_form_alter(array &$form, FormStateInterface $form_state) {
```

Comment:
Wrong function name. Needs to be commerce_cart_form_field_config_edit_form_alter. The current version never gets called :)